api/api_controller.go

```diff
@@ -202,6 +215,64 @@ func pageAmount(i *int64) int {
 	return inti
 }
 
+func (c *Controller) SetupLakeFSHandler() setupop.SetupLakeFSHandler {
+	return setupop.SetupLakeFSHandlerFunc(func(setupReq setupop.SetupLakeFSParams) middleware.Responder {
+		// skip migrate in case we have an active installation
+		if _, err := c.deps.Meta.InstallationID(); err == nil {
+			return setupop.NewSetupLakeFSConflict().
+				WithPayload(&models.Error{
+					Message: "lakeFS already initialized",
+				})
+		}
+
+		ctx := setupReq.HTTPRequest.Context()
+		err := c.deps.Migrator.Migrate(ctx)
+		if err != nil {
+			return setupop.NewSetupLakeFSDefault(http.StatusInternalServerError).
+				WithPayload(&models.Error{
+					Message: err.Error(),
+				})
+		}
+
+		metadata, err := c.deps.Meta.Write()
+		if err != nil {
+			return setupop.NewSetupLakeFSDefault(http.StatusInternalServerError).
+				WithPayload(&models.Error{
+					Message: err.Error(),
+				})
+		}
+
+		c.deps.Collector.SetInstallationID(metadata["installation_id"])
+		c.deps.Collector.CollectMetadata(metadata)
+		c.deps.Collector.CollectEvent("global", "init")
+
+		if len(*setupReq.User.DisplayName) == 0 {
```

Comment:
Can we move this check to the start so we fail on validation before we setup installation ID